library/src/compute/calc.rs

```diff
@@ -441,6 +478,34 @@ pub fn log(
     Value::Float(result)
 }
 
+/// Calculate the natural logarithm of a number.
+///
+/// ## Example { #example }
+/// ```example
+/// #calc.ln(100)
+/// ```
+///
+/// Display: Logarithm
+/// Category: calculate
+/// Returns: float
+#[func]
+pub fn ln(
+    /// The number whose logarithm to calculate. Must be strictly positive.
+    value: Spanned<Num>
+) -> Value {
+    let number = value.v.float();
+    if number <= 0.0 {
+        bail!(value.span, "value must be strictly positive")
+    }
+
+    let result = number.ln();
+    if result.is_infinite() || result.is_nan() {
+        bail!(args.span, "the result is not a real number")
+    }
```

Comment:
Same as previous comment, it'll always return a real number. If the given number is really big, its natural logarithm will be way smaller, but if it is really near to `0` (but positive), it might give a really give number, approximated to `-inf`. Maybe that you wanted to check specifically this case?